src/test/java/bio/terra/service/policy/flight/CreateSnapshotPolicyStepTest.java

```diff
@@ -0,0 +1,78 @@
+package bio.terra.service.policy.flight;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.policy.model.TpsPolicyInput;
+import bio.terra.policy.model.TpsPolicyInputs;
+import bio.terra.service.policy.PolicyService;
+import bio.terra.service.policy.exception.PolicyConflictException;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.service.snapshot.flight.create.CreateSnapshotPolicyStep;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import java.util.UUID;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag("bio.terra.common.category.Unit")
+public class CreateSnapshotPolicyStepTest {
+  @Mock private PolicyService policyService;
+  @Mock private FlightContext flightContext;
+
+  private static final UUID SNAPSHOT_ID = UUID.randomUUID();
+  private static final TpsPolicyInput protectedDataPolicy =
+      PolicyService.getProtectedDataPolicyInput();
+  private static final TpsPolicyInputs policies =
+      new TpsPolicyInputs().addInputsItem(protectedDataPolicy);
+
+  private void mockFlightMap() {
+    FlightMap workingMap = new FlightMap();
+    workingMap.put(SnapshotWorkingMapKeys.SNAPSHOT_ID, SNAPSHOT_ID);
+    when(flightContext.getWorkingMap()).thenReturn(workingMap);
+  }
```

Comment:
Just curious, is there a reason that this doesn't run in a `@BeforeEach` or `@BeforeAll`?

Reply:
The mocking isn't needed in `testNoPolicyDoUndo`, so if this was in `@BeforeEach/BeforeAll` the mocked method would need to use "lenient" strictness. I ran into something similar in my last PR and Phil suggested splitting it up this way: https://github.com/DataBiosphere/jade-data-repo/pull/1454#discussion_r1188848552

Reply:
Ohhh!  Super helpful, especially as I start writing JUnit 5 tests.  Thanks for the reference!